core/src/main/java/com/scalar/db/transaction/consensuscommit/RecoveryHandler.java

```diff
@@ -35,22 +39,61 @@ public RecoveryHandler(
   // lazy recovery in read phase
   public void recover(Selection selection, TransactionResult result) {
     LOGGER.debug("recovering for {}", result.getId());
+
+    // as the result doesn't have before image columns, need to get the latest result
+    Optional<TransactionResult> latestResult;
+    try {
+      latestResult = getLatestResult(selection, result);
+    } catch (ExecutionException e) {
+      LOGGER.warn("can't get the latest result", e);
+      return;
+    }
```

Comment:
As commented, the `result` doesn't have before image columns due to the change, we need to re-read the record again to get before image columns.